cloud-config/README.md

```diff
@@ -1,6 +1,135 @@
+    Cluster Frontend            : app-pg-template.yml
+    Cluster Elasticsearch       : es-nodes-template.yml
+    
+    # Non Standard Templates
+    Instance with remote pg     : app-es-template.yml
+    Instance with remote pg/es  : app-template.yml
 
+    Open one of the templates above to compare with ./template-parts.  Each variable '%(varname)s' 
+    in the template has a matching file in ./template-parts.  Next is a way to view and save
+    assembled templates.
+
+    We can save Assembled Templates with the --save-configure bin/deploy argument.  It will 
+    automacitallly determine which template to used based on input arguments.
+
+    $ bin/deploy --save-config-name 20200430
+
+    ### Output
+    Created assembeled template
+           ./cloud-config/assembled-templates/20200430-app-es-pg.yml
+    Deploy with
+           $ bin/deploy --use-prebuilt-config 20200430-app-es-pg
+    Diff with on the fly assembly.  Does not deploy
+           $ bin/deploy --use-prebuilt-config 20200430-app-es-pg --diff-configs
+    ###
+
+
+#### Directories:
+    template-parts              : Pieces of the templates
+    run-scripts                 : Install scripts runcmd_* template parts
+    configs                     : Configuration files used in run-scripts, like apache, java, es
+    assembled-templates         : Saved.  These still contains Run Varialbes
+
+#### Helper Script
+    create-ami.py               : Create amis from deployed --ami-build in AWS
+
+#### Run_Variables
+    * Run variables are in /etc/environment file on the instance.  
+    * They are used in the run-scripts to configure the system and application builds
+    * /etc/environment is loaded into login/ssh sessions so you can echo them on the instance.
+    * The file will contain dupicate entries when deploying from an AMI.  Last ones are used.
+    
+    View them locally with --dry-run along with other info.  Does not deploy
+    $ bin/deploy --dry-run
+
+    Add options like --test, --release-candidate, or --candidate to see the differences in run vars.
+    The ROLE should change along with other variables like ENCD_INDEX_PRIMARY, ENCD_INDEX_VIS,
+    ENCD_INDEX_REGION.  The env vars are prefixed with ENCD_ as to not conflict with other env vars.
+
+## Live Deployments
+    Here we'll deploy some instances using the --full-build args to avoid needing amis.  Building
+    from scratch is easier for cloud config development.
+
+#### QA/Development demo: app-es-pg-template.yml
+    $ bin/deploy --full-build
+
+    ### Output
+    Deploying app-es-pg
+    $ bin/deploy --full-build
+    create instance and wait for running state
+    ####
+
+#### Demo Cluster(es-wait): es-nodes-template.yml and app-pg-template
+    $ export CLUSTER_NAME='encd-dev'
+    $ bin/deploy --full-build --cluster-name "$CLUSTER_NAME" --es-wait
+   
+    ### Output
+    Deploying es-nodes
+    $ bin/deploy --full-build --cluster-name "$CLUSTER_NAME" --es-wait
```

Comment:
is the full deploy command part of the output; this part was confusing because it's the same command as above

Reply:
Yea, it's console output....I'm not sure if there is a better way to notate that.  Or maybe not have it?

I added it because some commands, like building an es-nodes cluster outputs the --es-ip needed for the frontend.